### PR TITLE
Make testing easier by adding a disable() method

### PR DESCRIPTION
**Pull request by LaurentEsc**
This PR is related to issue #39 and is a suggestion to allow easy Validator mocking:

    HoneypotValidator::shouldReceive('validateHoneytime')
        ->once()
        ->andReturn(true);

I have updated the documentation.

### CODE REVIEW

**Comment by LaurentEsc** [comment]
Moving the `HoneypotFacade` to another namespace is a breaking change, probably not the best idea ...

**Comment by garygreen** [comment]
To be honest I think at this point it would make sense to just get rid of the `HoneypotValidator.php` and move those 3 functions into `Honeypot.php` and update the tests. That way there isn't a need for two facades which seems overcomplicated imo.

**Comment by garygreen** [comment]
Also I think I would prefer to see something like `Honeypot::enable()` and `Honeypot::disable()` as oppose to the suggestion of mocking the validator functions.

**Comment by LaurentEsc** [comment]
Something like this ? I also like `Honeypot::disable()` for testing.

File: src/Msurguy/Honeypot/HoneypotServiceProvider.php [hunk 1]
```diff
@@ -49,8 +49,14 @@ public function boot()
             $translator = $app['translator'];
 
             // Add honeypot and honeytime custom validation rules
-            $validator->extend('honeypot', 'Msurguy\Honeypot\HoneypotValidator@validateHoneypot', $translator->get('honeypot::validation.honeypot'));
-            $validator->extend('honeytime', 'Msurguy\Honeypot\HoneypotValidator@validateHoneytime', $translator->get('honeypot::validation.honeytime'));
+            $validator->extend('honeypot', function($attribute, $value, $parameters) {
```
Comment (garygreen) [inline src/Msurguy/Honeypot/HoneypotServiceProvider.php]: These don't need anonymous functions anymore. Just `Msurguy\Honeypot\Honeypot@validateHoneypot`

**Comment by garygreen** [comment]
Looks good to me. Tests pass?

File: README.md [hunk 2]
```diff
@@ -69,6 +69,28 @@ Please note that "honeytime" takes a parameter specifying number of seconds it s
 
 That's it! Enjoy getting less spam in your inbox. If you need stronger spam protection, consider using [Akismet](https://github.com/kenmoini/akismet) or [reCaptcha](https://github.com/dontspamagain/recaptcha)   
 
+## Testing
```
Comment (garygreen) [inline README.md]: This needs updating

**Comment by garygreen** [comment]
Once updated I would suggest to squash the commits.

**Comment by LaurentEsc** [comment]
The problem with using Class@Method instead of a closure is that is creates a new instance to validate, so disable() doesn't work.

**Comment by garygreen** [comment]
Gotcha. I'm pretty sure the validator resolves the class thru the IOC so in theory if you did `honeypot@method` it should work.

**Comment by LaurentEsc** [comment]
Indeed, seems to work.

**Comment by LaurentEsc** [comment]
Tests  didn't pass after I cloned:

1) Msurguy\Tests\HoneypotTest::test_get_honeypot_form_html
BadMethodCallException: Method Mockery_0_Msurguy_Honeypot_Honeypot::getFormHTML() does not exist on this mock object

**Comment by garygreen** [comment]
It should be `generate`, tests prob has been failing on that for a while.

**Comment by LaurentEsc** [comment]
Documentation updated and commits squashed.

**Comment by garygreen** [comment]
Looks good. :+1:

**Pull request merged**